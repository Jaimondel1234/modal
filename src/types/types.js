export const types = {
  authLogin: "[Auth] Login",
  authLogout: "[Auth] Logout",

  uiOpenModal: "[ui] Open Modal",
  uiCloseModal: "[ui] Close Modal",

  noteAddNew: "[notes] Add New Notes",
  noteDeleted: "[notes] Delete Note",
  noteSetActive: "[notes] Set Active Note",
  noteUpdate: "[notes] Update note",
  noteSetListMode: "[notes] Set List Mode ",
  noteReset: "[notes] Reset notes",
};
