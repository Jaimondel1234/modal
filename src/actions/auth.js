import { types } from "../types/types";

export const authLogin = (email, password) => ({
  type: types.authLogin,
  email,
  password,
});

export const authLogout = () => ({
  type: types.authLogout,
});
