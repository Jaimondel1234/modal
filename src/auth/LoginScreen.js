import React from "react";
import { useForm } from "../hooks/useForm";
import {
  FlexStyle,
  PanelStyle,
  ButtonStyle,
  TextStyle,
  InputStyle,
} from "../styles/core";
import styled from "styled-components";
import { useDispatch } from "react-redux";
import { authLogin } from "../actions/auth";

/** Style of input of login */
const LoginInputStyle = styled(InputStyle)`
  border-radius: 7px;
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 20px;
  max-width: 400px;
  padding: 12px;
`;

/*
  Component that displays the login screen
*/
export const LoginScreen = () => {
  const dispatch = useDispatch();
  const [formValues, handleInputChange] = useForm({
    email: "admin@gmail.com",
    password: "admin",
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(authLogin(email, password));
    localStorage.setItem("user", JSON.stringify({ email, password }));
  };

  return (
    <PanelStyle bg='#1E90FF' color='#ffff' borderRadius={10}>
      <TextStyle
        fontSize='60px'
        fontWeight='bold'
        textShadow='2px 2px 4px #000000'
        textAlign='center'
        pb='20px'
        pt='25vh'
      >
        Login
      </TextStyle>

      <FlexStyle
        as='form'
        alignItems='center'
        justifyContent='space-between'
        flexDirection='column'
        onSubmit={handleLogin}
      >
        <LoginInputStyle
          type='text'
          placeholder='Your email'
          name='email'
          autoComplete='off'
          value={email}
          onChange={handleInputChange}
        />
        <LoginInputStyle
          type='password'
          placeholder='Your password'
          name='password'
          autoComplete='off'
          value={password}
          onChange={handleInputChange}
        />
        <ButtonStyle
          p='12px'
          fontSize='25px'
          fontWeight='bold'
          maxWidth='400px'
          letterSpacing='3px'
          lineHeight='40px'
          type='submit'
        >
          Send
        </ButtonStyle>
      </FlexStyle>
    </PanelStyle>
  );
};
