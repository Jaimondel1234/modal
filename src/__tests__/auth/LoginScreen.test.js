import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { LoginScreen } from "../../auth/LoginScreen";
import { types } from "../../types/types";

describe("Test <LoginScreen/>", () => {
  const mockStore = configureStore([]);
  const initState = {};

  const store = mockStore(initState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <LoginScreen />
    </Provider>
  );
  test("it must be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("submitting the form, triggers the action to log", () => {
    wrapper.find("form").prop("onSubmit")({ preventDefault: () => {} });
    expect(store.dispatch).toHaveBeenCalledWith({
      type: types.authLogin,
      email: "admin@gmail.com",
      password: "admin",
    });
  });
});
