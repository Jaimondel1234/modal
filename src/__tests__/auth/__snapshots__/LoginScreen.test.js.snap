// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <LoginScreen/> it must be displayed correctly 1`] = `
<div
  className="core__PanelStyle-sc-1mkz20q-1 hOgIJy"
  color="#ffff"
>
  <div
    className="core__TextStyle-sc-1mkz20q-2 fNbuYA"
    fontSize="60px"
    fontWeight="bold"
  >
    Login
  </div>
  <form
    className="core__FlexStyle-sc-1mkz20q-3 fVdgrf"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="core__InputStyle-sc-1mkz20q-4 LoginScreen__LoginInputStyle-vknzfc-0 kivKiO iyzkpe"
      name="email"
      onChange={[Function]}
      placeholder="Your email"
      type="text"
      value="admin@gmail.com"
    />
    <input
      autoComplete="off"
      className="core__InputStyle-sc-1mkz20q-4 LoginScreen__LoginInputStyle-vknzfc-0 kivKiO iyzkpe"
      name="password"
      onChange={[Function]}
      placeholder="Your password"
      type="password"
      value="admin"
    />
    <button
      className="core__ButtonStyle-sc-1mkz20q-5 jgCAkM"
      fontSize="25px"
      fontWeight="bold"
      letterSpacing="3px"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
